### PR TITLE
SNAP Rasdaman access and datacube viz

### LINKED ISSUE #7: Interactive hypercube viz
Use interactive plotting library e.g. hvplot to visualize some data, with widgets to control what data is displayed.

### PR DESCRIPTION
**Pull request by kyleredilla**
This PR adds a single notebook and some additional utility functions. 

The notebook breaks down using WCS to get a datacube from Rasdaman, and then provides a couple of examples of plotting with it. To test, make sure you can run the notebook and check typos and such! Also think about potential future enhancements for this notebook or more future notebooks, and comment on them or make some issues.

I again did not update the environment file, but the additional libraries needed are:

```
hvplot
panel
jupyter_bokeh
```

closes #7

### CODE REVIEW

File: notebooks/snap_rasdaman_access.ipynb [hunk 1]
```diff
@@ -0,0 +1,2120 @@
...
+    "\n",
+    "All of the datasets hosted here can be thought of as \"data cubes\" or \"hyper cubes\", meaning data that is stored in a structure with 3 or more (usually more) dimensions. Most of these datasets have two spatial axes representing a regular spatial grid, and a temporal axis such as timestamp, month, year, or even decade, and similar temporal groupings over which some source data has been aggregated. What pushes many of these datasets beyond true \"cubes\" is that they offer various configurations of modeled data, where there are additional dimensions included for things like the source climate model used, the emissions scenario used for the particular model run, or the operation used for a temporal aggregation.\n",
+    "\n",
+    "These data cubes are referred to as \"coverages\" in the protocols we will be using to access this data, so we will stick with that term for the remainder of this notebook. Read more about the concept of a coverage at the [Open Geospatial Consotrium's website](https://ogcapi.ogc.org/coverages/overview.html). \n",
```
Comment (Joshdpaul) [inline notebooks/snap_rasdaman_access.ipynb:13]:
```suggestion
    "These data cubes are referred to as \"coverages\" in the protocols we will be using to access this data, so we will stick with that term for the remainder of this notebook. Read more about the concept of a coverage at the [Open Geospatial Consortium's website](https://ogcapi.ogc.org/coverages/overview.html). \n",
```

**Review by Joshdpaul — CHANGES_REQUESTED** [review]
Kyle, this notebook is an excellent walk through of how to craft a WCS query. (I will probably refer back to this in the future!) The two new util functions are a great addition that allow the user to really quickly fetch/fuse the data and metadata. Very useful! 

I like both the plots; they are very "to the point" and don't require a ton of code, which I think keeps the focus on the data. Which I think works well for this notebook, which is more about accessing and understanding the coverages/datacube structure and less about cartography/plotting.

Most of my comments were fixing typos but there was one regarding the user inputs in the 3x1 plot. It looks like only the month input is used, and the model and scenario are just hard-coded. Not sure which direction you want to steer that?

One question is whether we will be able to host the notebooks in such a way as to have the final interactive plot useable through a web browser? That would be a great feature. I think being able to choose the dataset coordinates via point-and-click and then see the variable results is an effective way of explaining/transmitting the multidimensional nature of the coverage. To me, the list of coordinates/variables in the `xarray.Dataset` preview is less intuitive and I appreciate having the visualization / clickable component here. Nice work!

PS) I needed to also install `lxml` in my environment to use the new util functions

File: notebooks/snap_rasdaman_access.ipynb [hunk 2]
```diff
@@ -0,0 +1,2120 @@
...
...Jnf2RuHCaej770DvbpBuvqwmG8Rwumr4QMH3JL0LZU60G9WYyEtvsXZs0uPoR4exuslCidlzEQABAABJREFUaegUOGuV+4CF2bMRKAOcNrCajulnGVw17c+DHCIi0MKVmXks7KSImTTVUAU4BtrHLB/uh+oMBYDOcTNncntlQ5/ZUWwdLLNI4EiFqm5AUxs4Baiwg4pjq0BzVEeJU2MiO3mNMJOnFALSS7WM2zh3Dj8157Eyed4hbve1el86JlND+icb65dT/srS1PdRK2ijT/Zm2jXJ2fyGnDYlUw0ga/ze5CanVJ0J7WaA3VCMRwpx5gyqPfWBa2+Rv3PH0uCC8YZMq/zGI7Jzq3xnvbHG7FqsDfTlSwCAWl0grzbE7rk+9oPkrGraI/zG/wfh2/8dhzd3mG539VoatiP89QWthxnOfEHLrfcvkPJmbRUGlvyZOEKlCXr7mqKGDtvKzpeUmG2jaKjq/Cr7XGu6V+YEHd49juvUSFNE0hLBQZ9d+gPDtjElxAab6hauna1xVGZYIYeuH39x3mjPLPPQYlkAIE8cuL4P1Bc5Buq33tJ3IXl+OlrKT14Y1qkc6Xq/vUXZbyuDmSc6p0sulYHs8whzao/XbUmNdeydWZ86HnJKBZp5To33UnMmURuFjFOS1PdrTXiCuv2BuA/63Tu7ymMbo3FpNS6twgtnyIm8Xo8UjDOz65HmXsV+5CnB3B6gvalSd20U/DZg2oUuLkV6OgvCdoK/dFV5IVJ3WVc1DOPrhhOpu8TueM6QjRPyyPmydzfYfes1dl/cYPvtLfY348zMRzFD7K8usHp5OfNoqCZXQHNin+j+ku5uqGd6e4u0vZ8d89qxS3CISFNAGicYdpm26wgTA/RhRO5z5T7AIbnJDz3/aczHxwsic6I+Ru4flBto5gmbYcndkfJCbrSqVCAIgHL1BDz24DRn6pUSQMS5keTEGlp/BAAddsjGQXOos8z8xVlzZQyMYhOeApqcyUGdEkmaODpA6YA+DL0fM4Ap/8dQJy3lkRiAJaX/kKwVaH1jIvfrQeNy256yvIe2pV/Gcw1lNOwFm1kMG5J/2fnFUvuhxaYcaJJI3ww7J7o19MUV9cfx62RSiPwZ8v0bAMD0mhrks2TE7bcow2beP9M5hSrHQERkp9YjK4NUMknTwNI0Pk5WRsEc7qHHt9XdTVxfAXD/juMc
```
[the remark this answers is not in the document]
Comment (kyleredilla) [reply notebooks/snap_rasdaman_access.ipynb:1985]: Fixed this, they were supposed to determine values for the first plot (which they do now)

[the remark this answers is not in the document]
Comment (kyleredilla) [reply notebooks/snap_rasdaman_access.ipynb]: Yeah I am confused by the way this suggestion is formatted, so I fixed this in [51dc8d8](https://github.com/ua-snap/ardac-toolbox/pull/14/commits/51dc8d8041843609f82b2688057da5a57f72c9d5).

**Comment by kyleredilla** [comment]
Thanks for the review @Joshdpaul ! I added all of your suggestions.

I agree that it would be ideal if the final hosting solution will allow the interactive plot to be usable through a web browser.

**Pull request merged**